src/Compilers/CSharp/Portable/Syntax/SyntaxFormatter.cs

```diff
@@ -127,8 +129,27 @@ public override SyntaxToken VisitToken(SyntaxToken token)
             }
         }
 
+        private SyntaxToken GetNextRelevantToken(SyntaxToken token)
+        {
+            // get next token, skipping zero width tokens except for end-of-directive tokens
+            var nextToken = token.GetNextToken(
+                t => SyntaxToken.NonZeroWidth(t) || t.Kind() == SyntaxKind.EndOfDirectiveToken, 
+                t => t.Kind() == SyntaxKind.SkippedTokensTrivia);
+
+            if (_consideredSpan.Contains(nextToken.FullSpan))
+            {
+                return nextToken;
+            }
+            else
+            {
+                return default(SyntaxToken);
+            }
+        }
+
         private SyntaxTrivia GetIndentation(int count)
         {
+            count = Math.Max(count - _initialDepth, 0);
```

Comment:
Same reasoning as before.  The indentation was considering ancestors outside the root, which made unexpected formatting when you tried normalizing a sub-tree that originated from an existing tree.